sdks/java/core/src/main/java/org/apache/beam/sdk/util/construction/SplittableParDoNaiveBounded.java

```diff
@@ -524,22 +526,15 @@ public String timerId(DoFn<InputT, OutputT> doFn) {
       public OutputReceiver<OutputT> outputReceiver(DoFn<InputT, OutputT> doFn) {
         return new OutputReceiver<OutputT>() {
           @Override
-          public void output(OutputT output) {
-            outerContext.output(output);
-          }
-
-          @Override
-          public void outputWithTimestamp(OutputT output, Instant timestamp) {
-            outerContext.outputWithTimestamp(output, timestamp);
-          }
-
-          @Override
-          public void outputWindowedValue(
-              OutputT output,
-              Instant timestamp,
-              Collection<? extends BoundedWindow> windows,
-              PaneInfo paneInfo) {
-            outerContext.outputWindowedValue(output, timestamp, windows, paneInfo);
+          public OutputBuilder<OutputT> builder(OutputT value) {
+            return WindowedValue.<OutputT>builder(
+                    outputBuilder ->
+                        outerContext.outputWindowedValue(
```

Comment:
Noting that the underlyiing ProcessContext (or other Context) does not benefit from the OutputBuilder pattern and needs to propagate metadata, either by refactor or just brute force.

Reply:
OK now I've gotten far enough that I see how I can reverse the layering so the ProcessContext in all the FnApiDoFnRunners and SimpleDoFnRunner can/should use the builder properly. To be completed... doing FnApiDoFnRunner last because it is the most work.